include/entity/light.h

```diff
@@ -0,0 +1,27 @@
+#ifndef ENTITY_LIGHT_H
+#define ENTITY_LIGHT_H
+
+#include "point.h"
+
+struct Light {
+    Point location;
+
+    struct Intensity {
+        int red;
+        int green;
+        int blue;
+
+    private:
+        int clamp(int value);
+    public:
+        void setRed(int r);
+        void setGreen(int g);
+        void setBlue(int b);
+        
+    } intensity;
```

Comment:
to pensando se nesse caso em específico vale a pena ter uma struct com esses 3 ints, ou fica melhor só declarar eles 3 diretos. Tem dois pontos aqui:
* deixar os 3 ints mesmo, pq aí diminui código e diminui tbm o tamanho da linha q a gnt precisa fazer pra acessar.
* usar isso, mas definir fora, em alguma outra classe, pra não ficar aninhado.

eu acho que a gente pode só deixar os 3 ints soltos na struct, nesse caso mesmo. Mas vê aí o que tu acha melhor. Só não gosto muito das structs aninhadas,